File: Data Structures & Algorithms/is-palindrome/submission-1.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        # Creating two pointers: one at the beginning, and
        # one at the end
        left, right = 0, len(s) - 1

        # Keep performing operations until we meet in the middle
        while left < right:
            # If the character at our left pointer is not
            # alpha numeric, move our left pointer up
            while left < right and not s[left].isalnum():
                left += 1

            # If the character at our right pointer is not
            # alpha numeric, move our right pointer down
            while left < right and not s[right].isalnum():
                right -= 1

            # If the character at our pointers are not the same, then
            # it is not a palindrome
            if s[left].lower() != s[right].lower():
                return False
            else:
                # Otherwise, move our pointers
                left += 1
                right -= 1

        return True

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // Setting up our two pointers
        int left = 0, right = numbers.size() - 1;

        // Begin the two pointers approach
        while (left < right) {
            // Calculating our current sum
            int total = numbers[left] + numbers[right];

            // If our sum matches the target, return the indices (one-indexed)
            if (total == target) {
                return {left + 1, right + 1};
            } else if (total < target) {
                // We need bigger numbers so move left pointer up
                left++;
            } else {
                // Otherwise, we need smaller numbers so move right pointer down
                right--;
            }
        }
    }
};
